File: frontend/src/types/student.ts
import { User } from './user';

export interface Student extends User {
  studentCode: string;
  fullName: string; // Required for student profile
  className?: string;
  avatarUrl?: string;
  faceEmbedding?: string;
}

export interface StudentResponse extends Student {}

export interface StudentCreationRequest {
  username: string;
  password?: string;
  studentCode: string;
  fullName: string;
  email?: string;
  className?: string;
  avatarUrl?: string;
  faceEmbedding?: string;
}

export interface StudentUpdateRequest {
  fullName?: string;
  email?: string;
  className?: string;
  avatarUrl?: string;
  faceEmbedding?: string;
}

File: gian_lan/src/main/java/com/idai/gian_lan/entity/Student.java
package com.idai.gian_lan.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "students")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Student extends User {

    @Column(name = "student_code", unique = true, nullable = false)
    String studentCode;

    String className;
    String avatarUrl;

    @Column(columnDefinition = "TEXT")
    String faceEmbedding;
}
